fuseimg/data/ops/ops_mri.py

```diff
@@ -0,0 +1,981 @@
+                    spacing:Tuple[int,int,int]=(1, 1, 3), mask_inx = -1,is_use_mask=True):
+    """
+    crop_lesion_vol crop tensor around position
+    :param vol: vol to crop
+    :param position: point to crop around
+    :param ref: reference volume
+    :param size: size in pixels to crop
+    :param spacing: spacing to resample the col
+    :param center_slice: z coordinates of position
+    :param mask_inx: channel index in which mask is located default: last channel
+    :param is_use_mask: use mask to define crop bounding box
+    :return: cropped volume
+    """
+
+    margin = [20,20,0]
+    vol_np = sitk.GetArrayFromImage(vol)
+    if is_use_mask:
+
+        mask = sitk.GetArrayFromImage(vol)[:,:,:,mask_inx]
+        mask_bool = np.zeros(mask.shape).astype(int)
+        mask_bool[mask>0.01]=1
+        mask_final = sitk.GetImageFromArray(mask_bool)
+        mask_final.CopyInformation(ref)
+
+        lsif = sitk.LabelShapeStatisticsImageFilter()
+        lsif.Execute(mask_final)
+        bounding_box = np.array(lsif.GetBoundingBox(1))
+        vol_np[:, :, :, mask_inx] = mask_bool
+    else:
+        bounding_box = np.array([int(position[0]) - int(size[0] / 2),
+                       int(position[1]) - int(size[1] / 2),
+                       int(position[2]) - int(size[2] / 2),
+                       size[0],
+                       size[1],
+                       size[2]
+                       ])
+    # in z use a fixed number of slices,based on position
+    bounding_box[-1] = size[2]
+    bounding_box[2] = int(position[2]) - int(size[2]/2)
+
+    bounding_box_size = bounding_box[3:5][np.argmax(bounding_box[3:5])]
+    dshift = bounding_box[3:5] - bounding_box_size
+    dshift = np.append(dshift,0)
+
+    ijk_min_bound = np.maximum(bounding_box[0:3]+dshift - margin,0)
+    ijk_max_bound = np.maximum(bounding_box[0:3]+dshift+[bounding_box_size,bounding_box_size,bounding_box[-1]] + margin,0)
+
+
+
+    vol_np_cropped = vol_np[ijk_min_bound[2]:ijk_max_bound[2],ijk_min_bound[1]:ijk_max_bound[1],ijk_min_bound[0]:ijk_max_bound[0],:]
+    vol_np_resized = np.zeros((size[2],size[0],size[1],vol_np_cropped.shape[-1]))
+    for si in range(vol_np_cropped.shape[0]):
+        for ci in range(vol_np_cropped.shape[-1]):
+            vol_np_resized[si,:,:,ci] = cv2.resize(vol_np_cropped[si, :,:, ci], (size[0],size[1]), interpolation=cv2.INTER_AREA)
+
+    img = sitk.GetImageFromArray(vol_np_resized)
+    return img
+
+
+
+def extract_mask_from_annotation(vol_ref, bbox_coords):
+    xstart = bbox_coords[0]
+    ystart = bbox_coords[1]
+    zstart = bbox_coords[2]
+    xsize = bbox_coords[3]
+    ysize = bbox_coords[4]
+    zsize = bbox_coords[5]
+
+    mask = get_zeros_vol(vol_ref)
+    mask_np = sitk.GetArrayFromImage(mask)
+    mask_np[zstart:zstart + zsize, ystart:ystart + ysize, xstart:xstart + xsize] = 1.0
+    return mask_np
+
+
+def apply_rescaling(img: np.array, thres: tuple = (1.0, 99.0), method: str = 'noclip'):
```

Comment:
We have few Ops that doing rescaling,
Just choose one method and either use an existing implementation or create an op